### ISSUE TITLE
Allow passing of selected entity_id into action scripts

### ISSUE THREAD

**Issue by chriswood1001** [nielsfaber/scheduler-card#542]
### Checklist

- [X] I checked for similar existing requests (both open and closed) before posting.
- [X] My request is generic, other users may benefit from it too.
- [X] I will participate in further discussion about this feature and can test it (if requested) once it's done.

### Proposal

Much like 'entity' can be used in display options, also allow the selected 'entity' to be referred to when calling a script.

Example use cases are:
  * Create a "turn on child 1 lights at 7AM" schedule for which we select "light.child1" to turn on at 7AM that calls "script.wake_up" passing along the "light.child1" entity_id. We can re-use this wake_up script for "turn on child 2 lights at 7AM".
  
  * Multi-zone HVAC homes can select "climate.upstairs" to call "script.warm_and_do_stuff" at 6AM which passes along the "climate.upstairs" entity ID. We can use this same script for "climate.downstairs" at 7AM.

  * Lawn irrigation can call "script.zone_lights_and_water_on" for "input_boolean.lawn_zone1" at 9PM, then use this same script for zone2 at 9:30 and zone3 at 10:00.

### Additional info

Propose to expose the Scheduler's selected entity as a variable, such as `'{{ entity_id }}` for use in script. In the case of multiple scheduler entities, expose them all as an array.

```
customize:
  light.*:
    actions:
      - service: script.my_script
        service_data:
          my_entity: '{{entity_id}}' # The ID of the entity selected from Scheduler > Group > Entity
        name: My script
```

**Comment by nielsfaber** [comment]
The card is designed with the flow 'choose group/type' -> 'choose entity' -> 'choose action'.
Without customisations applied, in your example the corresponding click path would be 'scripts' -> 'script.my_script' -> 'run' (and I believe you will find be able to find your script back here).
I am well aware that it's possible to apply this 'hack' to make the script turn up as an action for a light, but this is not the intended usage.

More importantly, in the context of the scheduler card, the selected entity becomes `script.my_script` and the service becomes `script.my_script` as well. IMO this is a bit of a quirk of HA, I'd say it would be more consistent if the service was `script.execute` or `script.run` since this effectively describes the action performed on the entity (similar to light.turn_on etc.).

In your example the light entity has no relation with the script, you just want to pass it as extra parameter in the service_data (since you designed the script to accept it as input).
For this usage, the `variables` configuration is intended instead.
The way the card was designed, the assignment of variables is designed to be completely separated from the targeted entity and corresponding service, in the GUI this occurs in completely different pages.
I see no (elegant) way to make a 'bridge' in between such that the selected entity on the first step can be used as input on the second (in the form of a wildcard as you suggest, or differently).

In summary:
1.  The requested extra functionality has benefit for scripts only, specifically a script which requires an entity_id passed as parameter
2. The functionality requires users to customize the card to have services for scripts being mapped under different entity type (which I don't recommend)
3. I don't see a straight-forward way to adapt the code to facilitate this.

I think is that this is a very specific request which very few users will make use of, therefor I'm reluctant to spend time on it.
Hope you understand my view, of course you (and others!) are welcome to debate this.

**Comment by chriswood1001** [comment]
I appreciate you taking the time to evaluate the feature request and provide your explanation.
